rfcs/01924_tls_attestation_ca.md

```diff
@@ -0,0 +1,169 @@
+# RFC#01924: Remote Attestation via TLS (Certificate Authority)
+
+## Objective
+
+Remotely attest applications in Trusted Execution Environments (TEEs) over the
+TLS protocol.
+
+## Background
+
+A lot of existing applications that would benefit from the remote attestation
+use **TLS protocol** to create secure channels with their clients. But
+integrating the remote attestation might bring up unnecessary complications in a
+form of completely rewriting the application connection logic.
+
+Thus, instead of refactoring existing applications and implementing a new
+protocol, we propose to integrate remote attestation into TLS by
+cryptographically binding **TLS certificates** to the **TEE firmware**.
+
+## Overview
+
+In order to allow applications to use TLS for remote attestation, we propose to
+use a **Certificate Authority (CA)** that generates signed TLS certificates for
+TEE applications. The main idea is that instead of attesting TEE applications
+themselves, clients attest the CA, which attests TEE applications on behalf of
+the clients. Clients can then download the CA’s **root certificate** and use it
+for connecting to TEE applications via TLS.
+
+<!-- From: -->
+<!-- https://docs.google.com/drawings/d/1PUHf7NW3zV-zPz6zwTGUZI6wBVxy7yIfzxsbUbCw724 -->
+<img src="images/tls_attestation_ca.png" width="850">
+
+When a TEE application starts, the CA remotely attests it and creates a **signed
+TLS certificate** for it, which a TEE application can then use to create TLS
+channels with clients. Such certificates are signed by the CA’s root
+certificate.
+
+While generating a TLS certificate, the CA embeds the **TEE report** (which is a
+data structure signed by the TEE platform that contains information identifying
+the code that is running inside the TEE) into it, thus binding the TLS
+certificate to the TEE firmware. This way clients can check what payload is
+being run inside the TEE they are connecting to.
+
```

Comment:
There may be value in not embedding the full **TEE report** instead providing something that is easier for a client with limited resources to parse - perhaps just certain elements of the report (I'm thinking here about a Cortex-M client, that has on the order of 10s of kilobytes of RAM/Flash)